crypto/src/main/java/org/springframework/security/crypto/bcrypt/BCryptPasswordEncoder.java

```diff
@@ -109,6 +113,10 @@ public String encode(CharSequence rawPassword) {
 	}
 
 	public boolean matches(CharSequence rawPassword, String encodedPassword) {
+		if (rawPassword == null) {
+			throw new IllegalArgumentException("Raw password cannot be null");
```

Comment:
To follow Spring Security conventions, please change `Raw password` to match the variable name of `rawPassword`.

Reply:
Please add a test for this use case

Reply:
done